### PR TITLE
ONYX-10990 JWT mandatory and mapping feature

### PR DESCRIPTION
**Pull request by sashaCher**
### What does this PR do?

Combines the result of multiple PRs: #2326 , #2330 , #2331 :

Integration mapping and mandatory configuration into main JWT flow.

Basic sanity automation tests.

### What ticket does this PR close?
ONYX-10520, ONYX-10850

#### Change log
- [X] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [X] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [X] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [X] The changes in this PR do not affect the Conjur API

### CODE REVIEW

File: app/domain/logs.rb [hunk 1]
```diff
@@ -668,6 +668,11 @@ module AuthnJwt
         msg: "Successfully validated mapping claims configuration",
         code: "CONJ00132D"
       )
+
+      ClaimMapUsage = ::Util::TrackableLogMessageClass.new(
+        msg: "Checking restriction '{0-annotation-value}', fetching value from '{1-claim-name}' claim...",
```
Comment (sashaCher) [inline app/domain/logs.rb:673]: @hilagross @shulifink
Please review

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,8 +6,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
-### Changed
-- Conjur now returns 401 (instead of 500) for authentication requests with an empty username. [cyberark/conjur#2381](https://github.com/cyberark/conjur/issues/2318)
+### Added
+- Add mandatory claims support to JWT generic vendor configuration. ONYX-10520
```
[the remark this answers is not in the document]
Comment (orenbm) [reply CHANGELOG.md:10]: is this how we add changelog entries in Conjur now? without links to Jira? (i really don't know)

cc: @daphnaab

**Comment by tzheleznyak** [comment]
We need to add section about mapping in the CHANGELOG

**Comment by tzheleznyak** [comment]
image

**Comment by tzheleznyak** [comment]
image

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,8 +6,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+- Add claims mapping support to JWT generic vendor configuration. ONYX-10850
```
Comment (sashaCher) [inline CHANGELOG.md:11]: @tzheleznyak

Comment (tzheleznyak) [reply CHANGELOG.md:11]: Ohh sorry my mistake

**Review by tzheleznyak — APPROVED** [review]
LGTM

**Review by benoaviram — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hilagross) [reply CHANGELOG.md:10]: We should add a link to jira
@sashaCher @orenbm

**Pull request merged**